### ISSUE TITLE
soversion issues

### ISSUE THREAD

**Issue by EvaSDK** [rrthomas/enchant#153]
While upgrading the package for Gentoo, I noticed that 1.6 releases had soversion = 1.6.1 while release 2.1 has 0.2001.0. This is not too bad as the two soversion do denote an ABI break but the problem is that it seems backwards and is likely to cause problems on next ABI change if enchant 1.6 is still present.

**Comment by Arfrever** [comment]
Actually soname was previously `libenchant.so.1` and is now `libenchant.so.0`.

Anyway I think it would be reasonable if Enchant 2 was using soname `libenchant.so.2`.

**Comment by rrthomas** [comment]
Thanks for pointing this out. I have a fix ready that makes the sonum the same as the major version number.

**Comment by rrthomas** [comment]
@Arfrever, @EvaSDK, does either of you have a view as to the version number of the next release? I was going to use 2.1.1, but I wondered whether, given the soversion change, you thought it should be a bigger increment? I was thinking that since the 2.x releases are all recent and not widely packaged in e.g. long-term support distros, it probably doesn’t matter.

**Comment by Arfrever** [comment]
`2.1.1` as next version is fine.

If I backport this commit to `2.1.0`, then soname of library is `libenchant.so.2` and full name of library is `libenchant.so.2.2001.0`.
`.2001.0` part is irrelevant, but maybe full name as `libenchant.so.2.1.0` would be better?

**Comment by rrthomas** [comment]
PR #157 both does this, and, as far as I can tell, matches libtool's versioning rules better, which is odd (see the commit message for more info).

Assuming this looks OK once merged, I will bump the version and make a release.

**Comment by EvaSDK** [comment]
I don't remember for sure if libvirt even had ABI break but they would have the same trouble if they did.
This pattern of soversion also exists in gtk+ or libvirt-glib but they avoid the issue by changing the soname when doing major API changes like removing functions. They stick hard to backward compatibility throughout a major number series of releases and so never have this problem. Here are actual library filenames of these examples on my system:
/usr/lib64/libvirt.so.0.3006.0
/usr/lib64/libvirt-glib-1.0.so.0.1000.0
/usr/lib64/libgtk-3.so.0.2200.19
/usr/lib64/libgtk-x11-2.0.so.0.2400.31

In your case, if you stick to mostly the same rules as gtk+ you can probably just arrange to start at .so.2.2001.0 for 2.1.0.

**Comment by rrthomas** [comment]
Thanks both. I've switched to library filename linked to the version, since the version actually follows libtool's rules, and the scheme of libvirt, oddly, as I say, does not: the rules for AGE say:

  6. If any interfaces have been removed or changed since the last
     public release, then set AGE to 0.

But in the libvirt scheme, when the major version is incremented (which means that APIs have been removed or changed), then AGE gets bigger (because AGE is SONUM + MAJOR_VERSION * 1000).
